nncf/auto/environment/quantization_env.py

```diff
@@ -0,0 +1,669 @@
+
+        if reward > self.best_reward:
+            self.best_reward = reward
+            prGreen('New best policy: {}, reward: {:.3f}, acc: {:.3f}, model_ratio: {:.3f}, model_size(mb): {:.3f}'.format(
+                self.strategy, self.best_reward, quantized_acc, cur_model_ratio, cur_model_size/8000000))
+
+        obs = self.get_normalized_obs(len(collected_strategy)-1, 
+                                        only_pred=(self.bw_assignment_mode is BitwidthAssignmentMode.STRICT))
+                                                    
+        done = True
+        return obs, reward, done, info_set
+
+    def set_next_step_prev_action(self, idx, action, only_pred=True):
+        if only_pred:
+            self.master_df.loc[self.master_df.index[self.master_df.is_pred][idx], 'prev_action'] = action
+        else:
+            self.master_df.loc[self.master_df.index[idx], 'prev_action'] = action
+
+    def get_normalized_obs(self, idx, only_pred=True):
+        if only_pred:
+            _df = self.master_df.loc[self.master_df.is_pred, self.state_list]
+        else:
+            _df = self.master_df.loc[self.master_df.index, self.state_list]
+        _df.loc[_df.index, self.state_list] =  self.state_scaler.transform(_df[self.state_list])
+        return _df.iloc[idx]
+
+    def apply_actions(self, strategy):
+        self.master_df['action']=self.strategy
+
+        # step_cfg = deepcopy(self.orig_compress_cfg)
+        step_cfg = self.config['compression']
+
+        if 'scope_overrides' not in step_cfg:
+            step_cfg['scope_overrides']={}
+
+        if 'ignored_scopes' not in step_cfg:
+            step_cfg['ignored_scopes']=[]
+
+        # TODO: Handling for keeping layer at FP32
+        for layer in self.master_df.index:
+            precision=self.master_df.loc[layer, "action"]
+
+            if self.master_df.loc[layer, "is_wt_quantizer"]:
+                ScopeStr = str(self.master_df.loc[layer, 'state_scope'])
+                step_cfg['scope_overrides'][ScopeStr] = {}
+                step_cfg['scope_overrides'][ScopeStr]['bits']=int(precision) # int requires to convert numpy.int64 to int
+            else:
+                # if we use scope to non-weight quantizer, we would risk masking out the quantizers within the scope
+                # e.g. MobileNetV2/adaptive_avg_pool2d_0 masks all quantizers into the same scope
+                IAOpCtxStr = str(self.master_df.loc[layer, 'qid_obj'].ia_op_exec_context)
+                step_cfg['scope_overrides'][IAOpCtxStr] = {}
+                step_cfg['scope_overrides'][IAOpCtxStr]['bits']=int(precision)
+
+            self.master_df.loc[layer, "qmodule"].num_bits = precision # Actual actor to change quantizer precision
+        self.config['compression']=step_cfg
+        return True 
+        
+    def _get_state_space(self, quantization_controller, quantized_model, quantizer_table):
+        # TODO: can we use nncf utility to the following? like dummy forward?
```

Comment:
yes, just call:
```
quantized_model.do_dummy_forward()
```

Reply:
@ljaljushkin yes, just done it an hour ago. Have not pushed yet. :)